docs/guides/using-org-slugs.mdx

```diff
@@ -0,0 +1,397 @@
+  organization indicated by the slug. To achieve this, you can define two `organizationPatterns`: one for the
+  root (e.g. `/orgs/acmecorp`), and one using the wildcard matcher `(.*)` that will match
+  `/orgs/acmecorp/any/other/resource`.
+
+  You can use the same approach for the [personal account](https://clerk.com/glossary#personal-account) with `personalAccountPatterns`.
+
+  > [!WARNING]
+  > It may not always be possible to activate the organization requested in the URL. This can happen if no organization
+  > with the given slug exists, or the user isn't a member of the given organization. In these cases, clerkMiddleware
+  > **will not modify** the active organization, leaving as active whichever organization was previously active on the
+  > clerk session.
+
+  See the documentation for [organizationSyncOptions](/docs/references/nextjs/clerk-middleware#organization-sync-options)
+  for more specific usage of the middleware options.
+
+  ```tsx {{ filename: 'middleware.ts' }}
+  import { clerkMiddleware } from '@clerk/nextjs/server'
+
+  export default clerkMiddleware(
+    (auth, req) => {
+      if (isProtectedRoute(req)) auth().protect()
+    },
+    {
+      organizationSyncOptions: {
+        organizationPatterns: ['/orgs/:slug', '/orgs/:slug/(.*)'],
+        personalAccountPatterns: ['/me', '/me/(.*)'],
+      },
+    },
+  )
+
+  export const config = {
+    matcher: [
+      // Skip Next.js internals and all static files, unless found in search params
+      '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
+      // Always run for API routes
+      '/(api|trpc)(.*)',
+    ],
+  }
+  ```
+
+  > [!TIP]
+  > If you prefer to have an [organization ID](/docs/references/javascript/organization/organization#properties)
+  > appear in your URL rather than slugs, you can use `:id` placeholder instead of `:slug` in both `organizationPatterns`
+  > and `personalAccountPatterns`.
+
+  ### Server Rendered Organization Page
+
+  Now that clerkMiddleware is configured to activate organizations, you can build an organization-specific home page.
+
+  First, you must handle the case where the middleware cannot activate the organization based on the URL. This can
+  happen if no organization with the given slug exists, or if the given user isn't a member of the organization.
+
+  When this happens, the middleware will not change the active organization, so whatever organization was previously
+  active will remain active. To help with troubleshooting, a message will also be emitted by the middleware to the server
+  log:
+
+  > Clerk: Organization activation handshake loop detected. This is likely due to an invalid organization ID or slug.
+  > Skipping organization activation.
+
+  > [!CAUTION]
+  > It is ultimately the responsibility of the page to ensure that it renders the appropriate content for a given URL, and
```

Comment:
IMO those "caution" blocks can be a bit alarming to developers. I'd suggest removing this block and putting more specific guidance with comments on the code block below.

Reply:
I'll be honest, inspiring mild alarm in the reader is what I was going for. If a developer were to just look at the organization slug in the URL, assume this meant that the middleware authorized access to the org, then render some sensitive org-specific content from their backend, that would be a very bad bug.

Hey @BRKalow , do you happen to have an opinion on the best way to communicate this one?